test/mpi/rma/Makefile.am

```diff
@@ -425,6 +441,54 @@ putpscw1_SOURCES = putpscw1.c
 win_dynamic_acc_collattach_CPPFLAGS = -DUSE_INFO_COLL_ATTACH $(AM_CPPFLAGS)
 win_dynamic_acc_collattach_SOURCES = win_dynamic_acc.c
 
+win_dynamic_put_flush_get_CPPFLAGS = -DTEST_PUT $(AM_CPPFLAGS)
+win_dynamic_put_flush_get_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_acc_flush_get_CPPFLAGS = -DTEST_ACC $(AM_CPPFLAGS)
+win_dynamic_acc_flush_get_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_get_acc_flush_get_CPPFLAGS = -DTEST_GACC $(AM_CPPFLAGS)
+win_dynamic_get_acc_flush_get_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_fop_flush_get_CPPFLAGS = -DTEST_FOP $(AM_CPPFLAGS)
+win_dynamic_fop_flush_get_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_cas_flush_get_CPPFLAGS = -DTEST_CAS $(AM_CPPFLAGS)
+win_dynamic_cas_flush_get_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_put_flush_get_2region_CPPFLAGS = -DTEST_PUT -DTEST_2REGION $(AM_CPPFLAGS)
+win_dynamic_put_flush_get_2region_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_acc_flush_get_2region_CPPFLAGS = -DTEST_ACC -DTEST_2REGION $(AM_CPPFLAGS)
+win_dynamic_acc_flush_get_2region_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_get_acc_flush_get_2region_CPPFLAGS = -DTEST_GACC -DTEST_2REGION $(AM_CPPFLAGS)
+win_dynamic_get_acc_flush_get_2region_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_put_flush_get_collattach_CPPFLAGS = -DTEST_PUT -DUSE_INFO_COLL_ATTACH $(AM_CPPFLAGS)
+win_dynamic_put_flush_get_collattach_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_acc_flush_get_collattach_CPPFLAGS = -DTEST_ACC -DUSE_INFO_COLL_ATTACH $(AM_CPPFLAGS)
+win_dynamic_acc_flush_get_collattach_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_get_acc_flush_get_collattach_CPPFLAGS = -DTEST_GACC -DUSE_INFO_COLL_ATTACH $(AM_CPPFLAGS)
+win_dynamic_get_acc_flush_get_collattach_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_fop_flush_get_collattach_CPPFLAGS = -DTEST_FOP -DUSE_INFO_COLL_ATTACH $(AM_CPPFLAGS)
+win_dynamic_fop_flush_get_collattach_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_cas_flush_get_collattach_CPPFLAGS = -DTEST_CAS -DUSE_INFO_COLL_ATTACH $(AM_CPPFLAGS)
+win_dynamic_cas_flush_get_collattach_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_put_flush_get_2region_collattach_CPPFLAGS = -DTEST_PUT -DTEST_2REGION -DUSE_INFO_COLL_ATTACH $(AM_CPPFLAGS)
+win_dynamic_put_flush_get_2region_collattach_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_acc_flush_get_2region_collattach_CPPFLAGS = -DTEST_ACC -DTEST_2REGION -DUSE_INFO_COLL_ATTACH $(AM_CPPFLAGS)
+win_dynamic_acc_flush_get_2region_collattach_SOURCES = win_dynamic_rma_flush_get.c
+
+win_dynamic_get_acc_flush_get_2region_collattach_CPPFLAGS = -DTEST_GACC -DTEST_2REGION -DUSE_INFO_COLL_ATTACH $(AM_CPPFLAGS)
+win_dynamic_get_acc_flush_get_2region_collattach_SOURCES = win_dynamic_rma_flush_get.c
+
```

Comment:
In stead of `#ifdef`s, can we achieve the same thing using a command line option? It will at least simplify the `Makefiile`.

Reply:
I don't want to add too many input parameter handling codes in the test program because bugs may come from input handling. To make the Makefile slightly simpler, I can make the `2region` option be controlled by input. But I don't want change others (e.g., RMA operation type, coll_attach hint)

Reply:
I was only thinking to turn the `-DTEST_GACC` into input option such as `win_dynamic_test -op=gacc` or even `win_dynamic_test -gacc`. I think it will make both the makefile and code cleaner. Making the op an argument also make it an easier decision if we decide to test multiple rma op in a single test, right?

Reply:
I don't think we need generate multiple rma ops in this single test. But I changed to use -op=<put,...> as you suggested.